lib/philomena_web/controllers/autocomplete/tag_controller.ex

```diff
@@ -5,7 +5,103 @@ defmodule PhilomenaWeb.Autocomplete.TagController do
   alias Philomena.Tags.Tag
   import Ecto.Query
 
-  def show(conn, params) do
+  def show(conn, %{"vsn" => "2"} = params), do: show_v2(conn, params)
+  def show(conn, params), do: show_v1(conn, params)
+
+  # Returns a list of tag suggestions for an incomplete term. Does a prefix search
+  # on the canonical tag names and their aliases.
+  #
+  # See the docs on `show_v1` for the explanation on the breaking change we made
+  # in the `v2` version.
+  defp show_v2(conn, params) do
+    with {:ok, term} <- extract_term_v2(params),
+         {:ok, limit} <- extract_limit(params) do
+      suggestions = search(term, limit)
+      json(conn, %{suggestions: suggestions})
+    else
+      {:error, message} ->
+        conn
+        |> put_status(:unprocessable_entity)
+        |> json(%{error: message})
+    end
+  end
+
+  defp extract_term_v2(%{"term" => term}) when is_binary(term) and byte_size(term) > 2 do
+    result =
+      term
+      |> String.downcase()
+      |> String.trim()
+
+    {:ok, result}
+  end
+
+  defp extract_term_v2(%{"term" => _}),
+    do: {:error, "Term is too short, must be at least 3 characters"}
+
+  defp extract_term_v2(_params), do: {:error, "Term is missing"}
```

Comment:
You can use this `extract_term` for the old API if you slightly alter its case statement